### ISSUE TITLE
[warning] silx/io/specfile/src/sftools.c

### ISSUE THREAD

**Issue by alexmarie78** [silx-kit/silx#2304]
warning on compilation : 

```
int
sfSetCurrent( SpecFile *sf, long index,int *error )
{
     ObjectList *list,
                *flist;
     SpecScan   *scan,
                *fscan;
     long        nbytes;
     long        fileheadsize,start;

    /*
     * If same scan nothing to do
     */
     if (sfSameScan(sf,index)) return(0);

    /*
     * It is a new scan. Free memory allocated for previous one.
     */
     freeAllData(sf);

    /*
     * Find scan
     */
     list = findScanByIndex(&(sf->list),index);

     if (list == (ObjectList *)NULL) {
         *error = SF_ERR_SCAN_NOT_FOUND;
         return(-1);
     }

    /*
     * Read full scan into buffer
     */
     scan = list->contents;

     if (sf->scanbuffer != ( char * ) NULL) free(sf->scanbuffer);

     sf->scanbuffer = ( char *) malloc(scan->size);

     if (sf->scanbuffer == (char *)NULL) {
         *error = SF_ERR_MEMORY_ALLOC;
         return(-1);
     }

     lseek(sf->fd,scan->offset,SEEK_SET);

     nbytes = read(sf->fd,sf->scanbuffer,scan->size);
     if ( nbytes == -1) {
         *error = SF_ERR_FILE_READ;
         return(-1);
     }
     if ( sf->scanbuffer[0] != '#' || sf->scanbuffer[1] != 'S') {
         *error = SF_ERR_FILE_READ;
         return(-1);
     }
     sf->scanheadersize = scan->data_offset - scan->offset;

    /*
     * if different file read fileheader also
     */
     if (!sfSameFile(sf,list)) {
        if (sf->filebuffer != ( char * ) NULL) free(sf->filebuffer);

        start        = scan->file_header;
        flist        = findFirstInFile(&(sf->list),scan->file_header);
        if (flist == (ObjectList *) NULL) {
            fileheadsize = 0;
            sf->filebuffersize = fileheadsize;
        }
        else
        {
            fscan        = flist->contents;
            fileheadsize = fscan->offset - start;
        }

        if (fileheadsize > 0) {
            sf->filebuffer = ( char *) malloc(fileheadsize);
            if (sf->filebuffer == (char *)NULL) {
               *error = SF_ERR_MEMORY_ALLOC;
                return(-1);
            }
            lseek(sf->fd,start,SEEK_SET);
            read(sf->fd,sf->filebuffer,fileheadsize);
            if ( nbytes == -1) {
               *error = SF_ERR_FILE_READ;
               return(-1);
            }
            sf->filebuffersize = fileheadsize;
        }
     }
     sf->scansize = scan->size;
     sf->current  = list;

     return(1);
}
```
The return value of read() and lseek() aren't checked and a warning comes up. 
Moreover the test 'if(nbytes == -1)' is testing the first call of read(), not the one it was supposed.

**Comment by t20100** [comment]
Yes, this needs to be fixed.
Pull request welcome.

**Comment by t20100** [comment]
Closed by PR  #2319
